src/cmake/cmake_uninstall.cmake.in

```diff
@@ -0,0 +1,35 @@
+# Copyright(c) 2006 to 2018 ADLINK Technology Limited and others
+#
+# This program and the accompanying materials are made available under the
+# terms of the Eclipse Public License v. 2.0 which is available at
+# http://www.eclipse.org/legal/epl-2.0, or the Eclipse Distribution License
+# v. 1.0 which is available at
+# http://www.eclipse.org/org/documents/edl-v10.php.
+#
+# SPDX-License-Identifier: EPL-2.0 OR BSD-3-Clause
+
+# The code below is inspired by https://gitlab.kitware.com/cmake/community/wikis/FAQ#can-i-do-make-uninstall-with-cmake
+
+cmake_minimum_required(VERSION 3.5)
+
+if(NOT EXISTS "@CMAKE_BINARY_DIR@/install_manifest.txt")
+  message(FATAL_ERROR "Cannot find install manifest: @CMAKE_BINARY_DIR@/install_manifest.txt")
+endif(NOT EXISTS "@CMAKE_BINARY_DIR@/install_manifest.txt")
+
+file(READ "@CMAKE_BINARY_DIR@/install_manifest.txt" files)
+string(REGEX REPLACE "\n" ";" files "${files}")
```

Comment:
I see potential for interesting results here if the file names or the installation directory contains newlines or semicolons ... but ... I think it is fair to promise that neither the filenames nor the default installation directory will contain tricky character sequences.

So on consideration, I'm ok with committing this code. (I would have expected better from cmake though.)

Reply:
@eboasson I think you have a valid point here. There is nothing to prevent a user to use an install path contain weird directory names (e.g., directory names including \n). First question would be if we can install successfully. If so, we should also uninstall correctly.

Reply:
@eboasson I did a quick check to see if you can install in a "weird" directory. What I did is the following

    $ mkdir "build\ntest" 
    $ cd "build\ntest"
    $ cmake -DCMAKE_BUILD_TYPE=Release ../src

Even though "build\ntest" is a valid directory, installation fails miserably.

Reply:
My point was actually that one has to configure it in a really weird way, one unlikely to happen unintentially, and that it therefore is not a big deal. But it is really nice that you have actually spent time checking how it (mis)behaves :)